src/ovs/aca_arp_responder.cpp

```diff
@@ -0,0 +1,415 @@
+  pos->second = arp_cfg_in->mac_address;
+  _arp_db_mutex.unlock();
+
+  return EXIT_SUCCESS;
+
+}
+int ACA_ARP_Responder::delete_arp_entry(arp_config *arp_cfg_in){
+  arp_entry_data stData;
+
+  if (_validate_arp_entry(arp_cfg_in)) {
+    ACA_LOG_ERROR("Valiate arp cfg failed! (ip = %s and vlan id = %u)\n",
+                  arp_cfg_in->ipv4_address.c_str(),arp_cfg_in->vlan_id);
+    return EXIT_FAILURE;
+  }
+
+  if (_arp_db->empty()){
+    ACA_LOG_WARN("%s","ARP DB is empty!\n");
+    return EXIT_FAILURE;
+  }
+
+  ARP_ENTRY_DATA_SET((arp_entry_data *)&stData, arp_cfg_in);
+
+  if (_search_arp_entry(stData).empty()) {
+    ACA_LOG_ERROR("Entry not exist! (ip = %s and vlan id = %u)\n",
+                  arp_cfg_in->ipv4_address.c_str(),arp_cfg_in->vlan_id);
+    return EXIT_SUCCESS;
+  }
+
+  _arp_db_mutex.lock();
+  _arp_db->erase(stData);
+  _arp_db_mutex.unlock();
+
+  return EXIT_SUCCESS;
+}
+
+string ACA_ARP_Responder::_search_arp_entry(arp_entry_data stData){
+  auto pos = _arp_db->find(stData);
+
+  if(pos == _arp_db->end()){
+    return string();
+  }
+  return pos->second;
+
+}
+
+void ACA_ARP_Responder::_validate_mac_address(const char *mac_string)
+{
+  unsigned char mac[6];
+
+  if (!mac_string) {
+    throw std::invalid_argument("Input mac_string is null");
+  }
+
+  if (sscanf(mac_string, "%hhx:%hhx:%hhx:%hhx:%hhx:%hhx", &mac[0], &mac[1],
+             &mac[2], &mac[3], &mac[4], &mac[5]) == 6) {
+    return;
+  }
+
+  if (sscanf(mac_string, "%hhx-%hhx-%hhx-%hhx-%hhx-%hhx", &mac[0], &mac[1],
+             &mac[2], &mac[3], &mac[4], &mac[5]) == 6) {
+    return;
+  }
+
+  // nothing matched
+  ACA_LOG_ERROR("Invalid mac address: %s\n", mac_string);
+
+  throw std::invalid_argument("Input mac_string is not in the expect format");
+}
+
+void ACA_ARP_Responder::_validate_ipv4_address(const char *ip_address)
+{
+  struct sockaddr_in sa;
+
+  // inet_pton returns 1 for success 0 for failure
+  if (inet_pton(AF_INET, ip_address, &(sa.sin_addr)) != 1) {
+    throw std::invalid_argument("Virtual ipv4 address is not in the expect format");
+  }
+}
+
+void ACA_ARP_Responder::_validate_ipv6_address(const char *ip_address)
+{
+  struct sockaddr_in sa;
+
+  // inet_pton returns 1 for success 0 for failure
+  if (inet_pton(AF_INET6, ip_address, &(sa.sin_addr)) != 1) {
+    throw std::invalid_argument("Virtual ipv6 address is not in the expect format");
+  }
+}
+
+int ACA_ARP_Responder::_validate_arp_entry(arp_config *arp_cfg_in)
+{
+  if (0 >= arp_cfg_in->mac_address.size()) {
+    throw std::invalid_argument("Input mac_string is null");
+  }
+
+  _validate_mac_address(arp_cfg_in->mac_address.c_str());
+
+  if (0 < arp_cfg_in->ipv4_address.size()) {
+    _validate_ipv4_address(arp_cfg_in->ipv4_address.c_str());
+  }
+
+  if (0 < arp_cfg_in->ipv6_address.size()) {
+    _validate_ipv6_address(arp_cfg_in->ipv4_address.c_str());
+  }
+
+  return EXIT_SUCCESS;
+}
+
+
+
+
+void ACA_ARP_Responder::_standardize_mac_address(string &mac_string)
```

Comment:
you can remove this function and usage of it since Alcor controller's mac address is now standardized.